helix-view/src/view.rs

```diff
@@ -160,15 +130,25 @@ impl View {
         self.docs_access_history.push(id);
     }
 
-    pub fn inner_area(&self) -> Rect {
-        // TODO add abilty to not use cached offset for runtime configurable gutter
-        self.area.clip_left(self.gutter_offset).clip_bottom(1) // -1 for statusline
+    pub fn inner_area(&self, doc: &Document) -> Rect {
+        self.area.clip_left(self.gutter_offset(doc)).clip_bottom(1) // -1 for statusline
     }
 
-    pub fn gutters(&self) -> &[(Gutter, usize)] {
+    pub fn gutters(&self) -> &[GutterType] {
         &self.gutters
     }
 
+    pub fn gutter_offset(&self, doc: &Document) -> u16 {
+        let mut offset = 0;
+        for gutter in self.gutters.iter() {
```

Comment:
This for loop could be replaced with an iter.map().sum()

Reply:
swapped for iterator equivalent